pkg/webhook/utils.go

```diff
@@ -111,3 +117,110 @@ func ValidateSingletonCreation(ctx context.Context, cli client.Reader, req *admi
 	return DenyCountGtZero(ctx, cli, resourceGVK,
 		fmt.Sprintf("Only one instance of %s object is allowed", req.Kind.Kind))
 }
+
+// ValidateConnectionAnnotation validates the connection annotation  "opendatahub.io/connections"
+// If the annotation exists and has a non-empty value, it validates that the value references
+// a valid secret in the same namespace. Additionally, it checks the secret's connection type
+// annotation and rejects requests with invalid configurations. (see allowedTypes)
+// If the annotation doesn't exist or is empty, it allows the operation.
+//
+// Parameters:
+//   - ctx: Context for the API call (logger is extracted from here).
+//   - cli: The controller-runtime reader to use for getting secrets.
+//   - decoder: The admission decoder to decode the request object.
+//   - req: The admission request being processed.
+//   - allowedTypes: List of allowed connection types for validation.
+//
+// Returns:
+//   - admission.Response: The validation result
+//   - bool: true if injection should be performed (only for known valid connection types)
+//   - *corev1.Secret: The validated secret (only valid when injection should be performed)
+//   - string: The connection type (only valid when injection should be performed)
+//   - *unstructured.Unstructured: The decoded object (only valid when validation passes)
+func ValidateConnectionAnnotation(ctx context.Context,
+	cli client.Reader,
+	decoder admission.Decoder,
+	req admission.Request,
+	allowedTypes []string,
+) (admission.Response, bool, *corev1.Secret, string, *unstructured.Unstructured) {
```

Comment:
looking at the number of returned values, it seems the method is doing more than just validating, so maybe it should be broken up in different parts ?